plugin/path_login.go

```diff
@@ -511,15 +511,42 @@ func (b *GcpAuthBackend) pathGceLogin(ctx context.Context, req *logical.Request,
 		Auth: auth,
 	}
 
-	if role.AddGroupAliases {
-		crmClient, err := b.CRMClient(req.Storage)
-		if err != nil {
-			return nil, err
+	if role.AddGroupAliases || role.AddGroupLabelAliases {
+		aliases := []*logical.Alias{}
+		if role.AddGroupAliases {
+			ancestryAliases := []*logical.Alias{}
+			crmClient, err := b.CRMClient(req.Storage)
+			if err != nil {
+				return nil, err
+			}
+			ancestryAliases, err = b.groupAliases(crmClient, ctx, metadata.ProjectId)
+			if err != nil {
+				return nil, err
+			}
+			aliases = append(aliases, ancestryAliases...)
 		}
-
-		aliases, err := b.groupAliases(crmClient, ctx, metadata.ProjectId)
-		if err != nil {
-			return nil, err
+		if role.AddGroupLabelAliases {
+			labels, err := b.groupLabelAliases(computeClient, ctx, metadata.ProjectId, metadata.Zone, metadata.InstanceName)
+			if err != nil {
+				return nil, err
+			}
+			for k, v := range labels {
+				if len(role.BlocklistGroupLabelAliases) > 0 {
+					for _, blocklistLabel := range role.BlocklistGroupLabelAliases {
+						if k != blocklistLabel {
```

Comment:
I believe that this behavior is incorrect. I think you want to create a flag that is initialized as false and iterates over the `BlocklistGroupLabelAliases` and sets it to true if the label is found in the blocklist. If at the end of that iteration, the value is false, then you add the alias.

As it stands, I believe this will add the alias multiple times for every case that the label is not a particular blocklist entry.

Reply:
@Valarissa is correct. We should only be appending once we've checked the entire slice.